src/one/nio/serial/DeserializeStream.java

```diff
@@ -69,7 +73,38 @@ public Object readObject() throws IOException, ClassNotFoundException {
             context = Arrays.copyOf(context, context.length * 2);
         }
 
-        return serializer.read(this);
+        try {
+            return serializer.read(this);
+        } catch (SerializerNotFoundException e) {
+            if (findSerializerInFields(serializer, e.getUid())) {
+                // rollback
+                this.address = prevAddress;
+                this.limit = prevLimit;
+                this.offset = prevOffset;
```

Comment:
DeserializeStream cannot be rolled back like this. Its state is more than just these 3 fields, and deserialization process may have side effects.